### ISSUE TITLE
I think there might be some problem in your padding_same_conv.py

### ISSUE THREAD

**Issue by Gasoonjia** [Oldpan/faceswap-pytorch#1]
Your padding_same_conv.py is a brilliant implementation for pytorch users needing padding=same. It really helps me and I cannot express anything but gratitude for your wonderful work.

However, I think there might be some mistakes in your implementation, especially in your conv2d_same_padding function.
You only calculate the number of elements padded for the col of the input, and just copy it to the row, which might be wrong when stride[0] != stride [1] and dilation[0] != dilation[1].
And the type of your initial value of stride and dilation might be list according to your following code but not the int you set.

**Comment by xigua314** [comment]
something wrong in your code！
    x = t.randn(2, 128, 100, 1)
    nc2 = Conv2d(128, 128, (2, 1)) #your code
    nc4 = nn.Conv2d(128, 128, (2, 1))
    nc2_re = nc2(x)
    nc4_re = nc4(x)
    print nc2_re.size() #(2, 128, 100, 2)
    print nc4_re.size() # (2, 128, 99, 1)
